programs/ziti-edge-tunnel/netif_driver/linux/tun.c

```diff
@@ -206,7 +206,7 @@ static void dns_update_systemd_resolve(const char* tun, unsigned int ifindex, co
     int s = run_command_ex(false, "systemd-resolve --status | fgrep  'DNS Domain' | fgrep -q '~.'");
     // set wildcard domain if any other resolvers set it.
     if (s == 0) {
-        run_command(SYSTEMD_RESOLVE " -i %s --set-doamin='~.'", dns_maintainer.tun_name);
+        run_command(SYSTEMD_RESOLVE " -i %s --set-domain='~.'", dns_maintainer.tun_name);
```

Comment:
i feel like this must fix 'some other bug'?

Reply:
yes. without this change, our dns server will not be consulted when all of the following conditions are met:

- on Linux
- when the dns maintainer detects that `systemd-resolve` should be used to update the system's resolver
- and the global resolver has the "wildcard" dns route

with that said, this is an unlikely set of circumstances. the dns maintainer looks for update methods in this order:

1. sd-bus direct API calls
2. /usr/bin/resolvectl
3. /usr/bin/systemd-resolve

so this will only make a difference when the api and resolvectl methods fail. I think we've written the exact versions up somewhere (maybe @qrkourier or @sabedevops can comment), but on any reasonably recent linux distro the sd-bus API will be selected.

Reply:
Sorry. My only point is that it's a different bug and not pertaining to windows is all.